tests/integ/test_deepcopy.py

```diff
@@ -387,9 +387,6 @@ def test_deep_nested_select(session):
     [
         lambda session_: session_.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"]),
         lambda session_: session_.sql("select 1 as a, 2 as b"),
-        lambda session_: session_.table(
```

Comment:
we don't need this change?

Reply:
this test cases use cache_result, the table will be auto cleaned in the middle of execution as it's not referenced.
@sfc-gh-aalam is onboard with the change

Reply:
hmm I don't get it, from code, it seems based_df shouldn't be gced, then if it's cleaned in the middle of execution, isn't unexpected?

Reply:
the clean up code is working expected.

if you check the table set up process, we don't keep reference of the originally created cache_result() table, instead we just use the table_name which is a string var.
```python
session_.table(
     session_.sql("select 1 as a, 2 as b").cache_result().table_name
)
```

in this way, no one is referencing the cache_result() table, so it gets cleaned up.

if we want to table not cleaned up, the test needs to be rewritten into the following format
```
cache_df = session_.sql("select 1 as a, 2 as b").cache_result()
session_.table(cache_df.table_name)
```

Reply:
yeah, we can remove this test.